backend/src/API/coffeeChatAPI.ts

```diff
@@ -145,61 +179,135 @@ export const checkMemberMeetsCategory = async (
+    } else if (category === 'a pm (not your team)') {
+      const isPm = otherMember.role === 'pm';
+      const notSameTeam = haveNoCommonSubteams(submitter, otherMember);
+      status = isPm && notSameTeam ? 'pass' : 'fail';
+      if (status === 'fail') {
+        if (!isPm && !notSameTeam) {
+          message = `${otherMember.firstName} ${otherMember.lastName} is not a PM and is on the same team as ${submitter?.firstName} ${submitter?.lastName}`;
+        } else if (!isPm) {
+          message = `${otherMember.firstName} ${otherMember.lastName} is not a PM`;
+        } else if (!notSameTeam) {
+          message = `${otherMember.firstName} ${otherMember.lastName} is a PM, but on the same team as ${submitter?.firstName} ${submitter?.lastName}`;
+        }
+      }
+    } else if (category === 'a tpm (not your team)') {
+      const isTpm = otherMember.role === 'tpm';
+      const notSameTeam = haveNoCommonSubteams(submitter, otherMember);
+      status = isTpm && notSameTeam ? 'pass' : 'fail';
+      if (status === 'fail') {
+        if (!isTpm && !notSameTeam) {
+          message = `${otherMember.firstName} ${otherMember.lastName} is not a TPM and is on the same team as ${submitter?.firstName} ${submitter?.lastName}`;
+        } else if (!isTpm) {
+          message = `${otherMember.firstName} ${otherMember.lastName} is not a TPM`;
+        } else if (!notSameTeam) {
+          message = `${otherMember.firstName} ${otherMember.lastName} is a TPM, but is on the same team as ${submitter?.firstName} ${submitter?.lastName}`;
+        }
+      }
+    }
+
+    // If otherMemberProperties doesn't exist, status should stay undefined
+    if (otherMemberProperties) {
+      if (category === 'a newbie') {
+        status = otherMemberProperties.newbie ? 'pass' : 'fail';
+        status === 'fail' &&
+          (message = `${otherMember.firstName} ${otherMember.lastName} is not a newbie`);
+      } else if (category === 'is/was a TA') {
+        status = otherMemberProperties.ta ? 'pass' : 'fail';
+        status === 'fail' &&
+          (message = `${otherMember.firstName} ${otherMember.lastName} was never a TA`);
+      } else if (category === 'major/minor that is not cs/infosci') {
+        status = otherMemberProperties.notCsOrInfosci ? 'pass' : 'fail';
+        status === 'fail' &&
+          (message = `${otherMember.firstName} ${otherMember.lastName} is a CS or Infosci major`);
+      }
+
+      // If submitterProperties doesn't exist, status should stay undefined
+      if (submitterProperties) {
+        if (category === 'from a different college') {
+          status = otherMemberProperties.college !== submitterProperties.college ? 'pass' : 'fail';
+          status === 'fail' &&
+            (message = `${otherMember.firstName} ${otherMember.lastName} is from the same college as ${submitter.firstName} ${submitter.lastName} (${otherMemberProperties.college})`);
+        }
+      }
+
+      if (category === 'a lead (not your role)') {
```

Comment:
I think...we should be able to rule out some people even if they haven't filled out the form. Reason is...some people are clearly not leads, so the warning should pop up anyways. If the lead didn't fill in the form for what kind of lead they are, then it's harder to tell, but say I coffee chat "Patricia Huang". Just because there isn't a form response doesn't mean I shouldn't deduce that the "a lead (not your role)" category would be invalid.